jpos/src/main/java/org/jpos/iso/AsciiPrefixer.java

```diff
@@ -86,12 +86,16 @@ public void encodeLength(int length, byte[] b) throws ISOException
     }
 
     @Override
-    public int decodeLength(byte[] b, int offset)
-    {
+    public int decodeLength(byte[] b, int offset) throws ISOException {
         int len = 0;
         for (int i = 0; i < nDigits; i++)
         {
-            len = len * 10 + b[offset + i] - (byte)'0';
+            byte d = b[offset + i];
+            if(d < '0' || d > '9')
```

Comment:
Should we use `Character.isDigit(d)` instead to better show the intention? 
(although it's pretty clear in your code).

Reply:
Hi Alejandro, thx for reactivity.

I think it's slightly better as it is.

Please note that Character.isDigit() allows not only ASCII digits but also:
- '\u0030' through '\u0039', ISO-LATIN-1 digits ('0' through '9')
- '\u0660' through '\u0669', Arabic-Indic digits
- '\u06F0' through '\u06F9', Extended Arabic-Indic digits
- '\u0966' through '\u096F', Devanagari digits
- '\uFF10' through '\uFF19', Fullwidth digits

That's probably not an issue while using it together with the rest of the system (because byte to string is hardcoded to use ISO-8859-1 encoding), but I think it's a good idea to be restrictive in the case someone uses the class in isolation.

BTW: also comment that in some JAVADOC it is stated that US_ASCII (assumed 7 bit) is used, but the library uses extended 8-bit character set in ISO-8859-1 codepage.

Reply:
Well I don't have any inclination towards one way or the other, but since d is a byte not a char it can never be grater than '\u00FF'.

Reply:
Both good points. No need to change it, will merge it as it is.

Reply:
@alcarraz so true! My comment was pointless then... I really don't care which option.
Finally: what about BcdPrefixer or EbcdicPrefixer? Do you plan to patch it? Do you want inputs on those?

Reply:
Beware of side-effects with BCD related stuff and some special characters used in track2 ('D')

Reply:
I think @jmones is talking only about the prefixers, so it shouldn't affect the interpreters

Reply:
Absolutely, I understand. I was just raising a flag and self reminder in case the fix could be attempted at the `ISOUtil` level.